text/0201-scope-attributes.md

```diff
@@ -0,0 +1,148 @@
+# Introduce Scope Attributes
+
+This OTEP adds attributes to the Scope.
+
+## Motivation
+
+There are a few reasons why adding Scope attributes is a good idea:
+
+- There are 2 known use cases where Scope attributes can solve specific problems:
+  - Add support for [Meter "short_name"](https://github.com/open-telemetry/opentelemetry-specification/pull/2422),
+    represented as an attribute of Meter's Scope.
+  - Add support for differentiating the type of data emitted from the scopes that belong
```

Comment:
Do we imagine moving some of the existing span attributes to be a scope attributes?
Things like:
- `db.system` (static for the instrumentation library)
- `messaging.system`
- `messaging.protocol`
- `messaging.protocol_version`

etc.

I guess it can be handy to sometimes store "Connection level attributes" in the scope, instead of repeating them for each new span, things like: `db.connection_string`, `net.peer.name` etc. The downside is that user will need to create a different instrumentation scope for each connection, with the same name version and schema but different attributes.

Reply:
I like this idea. And think that creating a new scope makes sense in those cases, so it fits. The only downside I see is repeating the "parent scope's" attribute so would want the ability to merge scope attributes, allowing the creation of the per-connection scope by merging those additional attributes with the instrumentation library scope.

I'd argue this is expanding the scope of the OTEP too though :). Maybe keeping it to adding attributes to scope and the necessary things like how merging happens when exported through non-OTLP protocols is best and these use-cases can be worked on after the addition is accepted.

Reply:
I think this is one of the Open Questions I have in the OTEP.
I don't know the answer. :-)

Reply:
Such attributes are not really attributes of the telemetry emitter. There was once the proposal of creating a "prepared span" API for such purposes: https://github.com/open-telemetry/opentelemetry-specification/issues/734.

Reply:
I think this OTEP leaves enough leeway to decide one way or another, so this decision is not a blocker for the OTEP move forward.